dynamiq/nodes/node.py

```diff
@@ -339,7 +339,9 @@ def validate_input_schema(self, input_data: dict[str, Any]) -> dict[str, Any] |
 
         if self.input_schema:
             try:
-                return self.input_schema(**input_data)
+                return self.input_schema.model_validate(
+                    input_data, context=kwargs | self.get_context_for_input_schema()
+                )
             except Exception as e:
                 raise RecoverableAgentException(f"Input data validation failed: {e}")
```

Comment:
Here we raise `RecoverableAgentException`, which may make sense for tools, but just that used in Agents. For other cases, we should raise default error